FPIS/Views/AddDesignation.cs

```diff
@@ -0,0 +1,197 @@
+            dbContext.Dispose();
+
+            if (designation != null)
+            {
+                _isDataValid = false;
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Designation already Created.";
+                return;
+            }
+
+            Regex regex = new(nameRegexPattern);
+
+            if (!regex.IsMatch(designationName))
+            {
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Designation name cannot have numbers.";
+                _isDataValid = false;
+                return;
+            }
+        }
+
+        private void ValidateInputs()
+        {
+            ValidateDepartmentName();
+            ValidateDesignationName();
+
+        }
+
+        private void LoadDeparments()
+        {
+            try
+            {
+                AppDbContext dbContext = new();
+                var departmentName = from Department in dbContext.Departments
+                                     select Department;
+                materialComboBoxDepartmentName.DataSource = departmentName.ToList();
+                materialComboBoxDepartmentName.DisplayMember = "DepartmentName";
+                // materialComboBoxDepartmentName.Text = "";
+                dbContext.Dispose();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error Loading Departments: {ex}");
+                MaterialMessageBox.Show(ex.ToString());
+            }
+        }
+
+        private void AddDesignation_Load(object sender, EventArgs e)
+        {
+           LoadDeparments();
+        }
+
+       
+        private void CreateDesignation()
+        {
+            btnAddDesignation.Enabled = false;
+            AppDbContext dbContext = new();
+            var departmentName = dbContext.Departments.Where(dpt => dpt.DepartmentName == materialComboBoxDepartmentName.Text).Single();
+            var department = dbContext.Departments.Single(dept => dept.Id == departmentName.Id);
+            
+            Designation designation = new()
+            {
+                
+                DesignationName = materialTextBoxDesignationName.Text,
+                DepartmentId = department.Id,
+
+            };
+
+            try
+            {
+                //AppDbContext dbContext = new();
+                dbContext.Designations.Add(designation);
+                dbContext.SaveChanges();
+
+                MessageBox.Show(
+                    $"{designation.DesignationName} Designation created successfully under {materialComboBoxDepartmentName.Text} Department.",
+                    "Success",
+                    MessageBoxButtons.OK,
+                    MessageBoxIcon.Information
+                );
+                materialComboBoxDepartmentName.Text = "";
+                materialTextBoxDesignationName.Text = "";
+                dbContext.Dispose();
```

Comment:
This line also has to move into the finally block. Just as I mentioned in the `AddDepartment.cs` file